crates/wasmtime/src/runtime/vm/libcalls.rs

```diff
@@ -1548,3 +1548,25 @@ fn cont_new(
         crate::vm::stack_switching::cont_new(store, instance, func, param_count, result_count)?;
     Ok(Some(AllocationSize(ans.cast::<u8>() as usize)))
 }
+
+#[cfg(feature = "gc")]
+unsafe fn get_instance_id(_store: &mut dyn VMStore, instance: Pin<&mut Instance>) -> u32 {
+    instance.id().as_u32()
+}
+
+#[cfg(feature = "gc")]
+unsafe fn throw_ref(
+    mut store: &mut dyn VMStore,
+    _instance: Pin<&mut Instance>,
+    exnref: u32,
+) -> Result<(), TrapReason> {
+    use crate::{AsStoreOpaqueMut, vm::ExceptionTombstone};
+
+    let exnref = VMGcRef::from_raw_u32(exnref).ok_or_else(|| Trap::NullReference)?;
+    let exnref = store.unwrap_gc_store_mut().clone_gc_ref(&exnref);
+    let exnref = exnref
+        .into_exnref(&*store.unwrap_gc_store().gc_heap)
+        .expect("gc ref should be an exception object");
+    store.as_store_opaque_mut().set_pending_exception(exnref);
+    Err(TrapReason::User(ExceptionTombstone.into()))
```

Comment:
Would it be possible to have `TrapReason::Exception` directly instead of, even at the lowest layers, encoding through the indirection of `anyhow::Error`?